medic/cht-core
# add GH Action to replace placeholders with staging urls

# Description

This PR adds an action that looks for 3 place holder values and then replaces them with deep links to download the compose files.  See [readme for more info](https://github.com/medic/cht-core/tree/7848-add-gh-action-for-staging-urls/.github/actions/update-staging-url-placeholders). 

per #7848


# Code review checklist


# Compose URLs

* __CHT_CORE_COMPOSE_URL__
* __COUCH_SINGLE_COMPOSE_URL__
* __COUCH_CLUSTER_COMPOSE_URL__

# testing
1. check out this branch `7848-add-gh-action-for-staging-urls`
2. edit [the workflow file on line 17](https://github.com/medic/cht-core/blob/4cd4f84968f72ee911d953f56d5378a4e3cb1f30/.github/workflows/replace-staging-urls.yml#L17) to have not be commented out
3. commit and push your  change
4. ensure workflow runs and the "Compose URLs" section above gets converted to URLs
5. IMPORTANT: comment back out line 17, commit and push so that  it defaults to `master` for normal runs
 
# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.
